pkg/virt-api/webhooks/utils.go

```diff
@@ -108,3 +108,17 @@ func IsPPC64(vmiSpec *v1.VirtualMachineInstanceSpec) bool {
 	}
 	return false
 }
+
+func IsRequestingVFIODevice(spec *v1.VirtualMachineInstanceSpec) bool {
+	var sriovVMI bool
+	hostDevVMI := spec.Domain.Devices.HostDevices != nil && len(spec.Domain.Devices.HostDevices) != 0
+	gpuVMI := spec.Domain.Devices.GPUs != nil && len(spec.Domain.Devices.GPUs) != 0
+	for _, iface := range spec.Domain.Devices.Interfaces {
+		if iface.SRIOV != nil {
+			sriovVMI = true
+			break
+		}
+	}
+
+	return sriovVMI || hostDevVMI || gpuVMI
+}
```

Comment:
I've refactored this, the only problem is that `IsHostDevVMI` and `IsGPUVMI` functions are used for VMI Resource Rules in `virt-controller/services/template.go` so I used the original functions as a wrapper for the updated function that accepts VMISpec instead of VMI as a parameter. If there is a way to implement it to take the VMISpec directly though without the extra functions that would be better, I'm not sure how to do it though.